### PR TITLE
[ISSUE-7099] Added failing test case for disappearing if-empty statement

### PR DESCRIPTION
**Pull request by dorrogeray**
Added failing test case for https://github.com/rectorphp/rector/issues/7099, it can be run with:

 ```sh
vendor/bin/phpunit --filter RuleCombinationShouldNotRemoveIfEmptyStmtRectorTest
```

### CODE REVIEW

File: tests/Issues/Issue7099/Fixture/fixture.php.inc [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+
+    public function __construct($data) {
+        $this->data = $data;
+    }
+
+    /** @return mixed */
+    public function getData() {
+        return $this->data;
+    }
+}
```
Comment (TomasVotruba) [inline tests/Issues/Issue7099/Fixture/fixture.php.inc]: The classes that are not being changed, but are only used to analyze, should be in own `/Source` directory. With correct namespace and PSR-4 autoloaded.

E.g. https://github.com/rectorphp/rector-src/tree/main/rules-tests/Renaming/Rector/PropertyFetch/RenamePropertyRector/Source

**Comment by dorrogeray** [comment]
@TomasVotruba I have refactored and simplified the test case. It seems that the ternary operator `$id === 1 ? new Document() : null;` is what "triggers" the problem in this simplified test case. :thinking:

**Comment by samsonasik** [comment]
@dorrogeray I cherry-picked your commit at PR https://github.com/rectorphp/rector-src/pull/2028

**Comment by TomasVotruba** [comment]
@dorrogeray Thank you 👍

**Pull request closed without merging**